makePoplogCommander.sh

```diff
@@ -29,51 +54,81 @@ cat << \****
 #endif
 
 //  Bit-flags.
-#define RUN_INIT_P          1
-#define INHERIT_ENV         2
-//  Bit-flag sets.
-#define PREFER_SECURITY     0
+#define RUN_INIT_P          0x1
+#define INHERIT_ENV         0x2
+#define VARIANT_X           0x4
+#define VARIANT_MOTIF       0x8
+//  Bit-flag sets for run vs dev.
+#define RUN_FLAGS           0x3
+#define PREFER_SECURITY     0x0
 #define PREFER_FLEXIBILITY  (RUN_INIT_P|INHERIT_ENV)
+//  Bit-flag sets for variants.
+#define VARIANT_FLAGS       0xC
+#define VARIANT_NOX         0x0
+#define VARIANT_XT          (VARIANT_X)
+#define VARIANT_XM          (VARIANT_X | VARIANT_MOTIF)
+#define VARIANT_UNSET       (VARIANT_MOTIF)
+#define INITIAL_FLAGS       (PREFER_FLEXIBILITY | VARIANT_UNSET)
 
 
 ///////////////////////////////////////////////////////////////////////////////
 //  Library routines: 
-//      Mishap, an error reporting function
-//      Chain, a 1D vector
+//      startsWith, a predicate on strings
+//      strEquals, another predicate on strings
+//      mishap, an error reporting function
+//      Vector, managed 1D vectors
+//      Deque, managed double ended queue
 ///////////////////////////////////////////////////////////////////////////////
 
+//  startsWith: does the subject start with prefix? ----------------------------
+bool startsWith( const char * subject, const char * prefix ) {
+    while ( *prefix ) {
+        if ( *prefix++ != *subject++ ) return false;
+    }
+    return true;
+}
+
+//  strEquals: are two strings equal? ------------------------------------------
+bool strEquals(const char * subject, const char * prefix ) {
+    return strcmp( subject, prefix ) == 0;
+}
+
 
-//  Mishap - Error reporting using printf-like functionality.
+//  Mishap - Error reporting using printf-like functionality. ------------------
 
 void mishap( const char *msg, ... ) {
     va_list args;
     va_start( args, msg );
     fprintf( stderr, "Mishap: " );
     vfprintf( stderr, msg, args );
-    fprintf( stderr, "\n" );
+    fprintf( stderr, "\n" );            // You do not have to supply the \n.
```

Comment:
Isn't that just repeating the code? I'd either remove it, or put it in the function comment on line 97.

Reply:
I will move the comment. I think it belongs with this function - but it should be a bit more expansive.